src/forms/SubmitButton.tsx

```diff
@@ -0,0 +1,30 @@
+import { ObjectState } from "@homebound/form-state";
+import { Button, ButtonProps, useComputed } from "src";
+
+export type SubmitButtonProps<T> = Omit<ButtonProps, "label"> & {
+  label?: ButtonProps["label"];
+  form: ObjectState<T>;
+};
+
+/** Provides a Button that will auto-disable if `formState` is invalid. */
+export function SubmitButton<T>(props: SubmitButtonProps<T>) {
+  const { form, disabled, onClick, label = "Submit", ...others } = props;
+  const invalidAndTouched = useComputed(() => !form.valid && form.touched, [form]);
+  return (
+    <Button
+      label={label}
+      disabled={disabled || invalidAndTouched}
```

Comment:
**suggestion:** I believe our existing pattern for "when to enable the button" was when the form is `dirty`. Then allowing the user to hit the submit button, which would either make the API call, or highlight invalid fields. I _think_ that still sounds like the right approach, wdyt?

Reply:
Yep, good call; I was staring right at our old `FormActions` doing `!isDirty` and just spaced out what it was doing. :-| :-D Updated!